### PR TITLE
DVOP-751 Evaluate and update GHA

### PR DESCRIPTION
**Pull request by palchyk-alex**
## Create GitHub Action to run build, test and publish.
This PR edits GitHub Action workflow file, for automated RocketChat build, test and publish process.

| **ClickUp** | **https://app.clickup.com/t/14247018/DEVOPS-2192** |
| ---------- | -------------------------------------------------------- |
| **Jira** | **https://drivecentric.atlassian.net/browse/DVOP-751** |

### CODE REVIEW

**Comment by palchyk-alex** [comment]
Before merging, need to change secrets in this repo

File: .github/workflows/build-release.yml [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: '3.1.x'
+      
+    - name: Install GitVersion
+      uses: gittools/actions/gitversion/setup@v0.9.9
+      with:
+        versionSpec: '5.6.7'
+        
+    - name: Get release version
```
Comment (JamesLoftis) [inline .github/workflows/build-release.yml:281]: This one is still using gitversion

File: .github/workflows/build-release.yml [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+    - name: Build Docker image for tag  
+      env:
+        GIT_TAG: ${{ steps.format-version.outputs.VERSION_STRING }}
+        
+      run: |
+        IMAGE="${{ steps.login-ecr.outputs.registry }}/${ECR_REPOSITORY}"
+        cd /tmp/build
+        docker build -t ${IMAGE}:$GIT_TAG .
+        docker push ${IMAGE}:$GIT_TAG
```
Comment (JamesLoftis) [inline .github/workflows/build-release.yml:320]: Need to add a script to create the octopus release to the end

**Review by JamesLoftis — APPROVED** [review]

**Comment by palchyk-alex** [comment]
## Changes summary

1. Added GitHub Actions files for releases and for PRs
2. Removed Jenkinsfile

## Release steps

1. Merge this **PR** to `develop`
3. Verify image creation in [MGMT accounts registry](https://us-east-1.console.aws.amazon.com/ecr/repositories/private/476851419095/rocketchat?region=us-east-1)

**Review by mlapaglia — DISMISSED** [review]

**Review by mlapaglia — APPROVED** [review]

**Review by JamesLoftis — APPROVED** [review]

**Review by dgibson-DriveCentric — APPROVED** [review]

**Pull request merged**